frame/bags-list/src/list/tests.rs

```diff
@@ -374,6 +366,127 @@ mod list {
 			assert!(non_existent_ids.iter().all(|id| !List::<Runtime>::contains(id)));
 		})
 	}
+
+	#[test]
+	#[should_panic = "given nodes must always have a valid. qed."]
+	fn put_in_front_of_panics_if_bag_not_found() {
+		ExtBuilder::default().build_and_execute_no_post_check(|| {
+			let node_10_no_bag = Node::<Runtime> { id: 10, prev: None, next: None, bag_upper: 15 };
+			let node_11_no_bag = Node::<Runtime> { id: 11, prev: None, next: None, bag_upper: 15 };
+
+			// given
+			ListNodes::<Runtime>::insert(10, node_10_no_bag);
+			ListNodes::<Runtime>::insert(11, node_11_no_bag);
+			StakingMock::set_vote_weight_of(&10, 14);
+			StakingMock::set_vote_weight_of(&11, 15);
+			assert!(!ListBags::<Runtime>::contains_key(15));
+			assert_eq!(List::<Runtime>::get_bags(), vec![(10, vec![1]), (1_000, vec![2, 3, 4])]);
+
+			let weight_fn = Box::new(<Runtime as Config>::VoteWeightProvider::vote_weight);
+
+			// then
+			assert_storage_noop!(assert_eq!(
+				List::<Runtime>::put_in_front_of(&10, &11, weight_fn).unwrap_err(),
+				crate::pallet::Error::<Runtime>::BagNotFound
+			));
+		});
+	}
+
+	#[test]
+	fn insert_at_unchecked_at_is_only_node() {
+		// Note that `insert_at_unchecked` tests should fail post checks because it does not
+		// increment the node counter.
+		ExtBuilder::default().build_and_execute_no_post_check(|| {
```

Comment:
```suggestion
		ExtBuilder::default().build_and_execute(|| {
```

probably stale comment -- it does now